### DISCUSSION TITLE
Action does not run on master with tags-ignore v*

### DISCUSSION THREAD

**Discussion by hauner** [community/community#26072] [Actions]
<p>I have a workflow that does not run on master after adding <code>tags-ignore</code> to <code>on/push</code>. When I remove the <code>tags-ignore</code> it does run again. When I re-add it it does not run.</p>
<p>I want to ignore tags so it does not run when I create a release tag.</p>
<p>Any idea why this is happening?</p>
<pre><code class="lang-auto">name: build

on:
  push:
    paths-ignore:
      - 'LICENSE'
      - 'README.adoc'
      - 'docs/**'
   tags-ignore:
     - v*
</code></pre>
<p>here is the workflow file:<br>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/openapi-processor/openapi-processor-core/blob/630ee1d91440047acc6be5e3db7eab9ec6d285c6/.github/workflows/build.yaml" target="_blank" rel="nofollow noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/openapi-processor/openapi-processor-core/blob/630ee1d91440047acc6be5e3db7eab9ec6d285c6/.github/workflows/build.yaml" target="_blank" rel="nofollow noopener">openapi-processor/openapi-processor-core/blob/630ee1d91440047acc6be5e3db7eab9ec6d285c6/.github/workflows/build.yaml</a></h4>
<pre><code class="lang-yaml">name: build

on:
  push:
#    paths:
#      - 'src/**'
#      - 'build.gradle'
#      - '.github/**'
    paths-ignore:
      - 'LICENSE'
      - 'README.adoc'
      - 'docs/**'
    tags-ignore:
      - v*

jobs:

  test:
    name: test
</code></pre>

  This file has been truncated. <a href="https://github.com/openapi-processor/openapi-processor-core/blob/630ee1d91440047acc6be5e3db7eab9ec6d285c6/.github/workflows/build.yaml" target="_blank" rel="nofollow noopener">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
</p>

**Comment by Simran-B** [comment]
<p>The asterisk is a special character in YAML. It is used in conjunction with <code>&amp;</code>, which defines an anchor, that can then be referenced it with <code>*</code>. I’m not sure if it causes your problem, as the string isn’t starting with an asterisk, but you should try to wrap the value in quotes:</p>
<pre><code class="lang-auto">    tags-ignore:
      - 'v*'
</code></pre>
<p><a href="https://yaml.org/spec/1.2/spec.html#*%20alias//" class="onebox" target="_blank" rel="nofollow noopener">https://yaml.org/spec/1.2/spec.html#*%20alias//</a></p>

**Comment by hauner** [comment]
<p>Thanks,</p>
<p>tried it, but unfortunately it does not help.</p>
<p>Looking at the docs, the tags are written without the quotes: <a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags" rel="nofollow noopener">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags</a></p>
<p>So that doesn’t seem to be the problem. Strange.</p>

**Comment by Simran-B** [comment]
<p>Weird. It should run whenever you push a commit to any branch, with at least one file not matching <code>paths-ignore</code>. You are not using a forked repo are you?</p>
<p>What if you remove <code>tags-ignore</code> completely and use a negated tag glob instead?</p>
<pre><code class="lang-auto">tags:
  - '!v*'
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags">The documentation</a> says:</p>
<blockquote>
<p>If you define only <code>tags</code> or only <code>branches</code> , the workflow won’t run for events affecting the undefined Git ref.</p>
</blockquote>
<p>Your workflow has an explicit rule for tags, but not for branches, so I assume that’s the reason. Try adding</p>
<pre><code class="lang-yaml">branches:
  - '**'
</code></pre>
<p>to the <code>push</code> event to build on all branch pushes.</p>

**Comment by hauner** [comment]
<p>Tried it, it does not help. After removing the tags stuff it runs again.</p>
<p>The repo is not forked, but I moved it to an <em>organization</em> a while ago.</p>

**Comment by hauner** [comment]
<p>Looks like that was the trick, now it runs! Thanks.</p>
